widgets/matplotlib/measurement/energy_spectrum.py

```diff
@@ -83,33 +87,48 @@ def __init__(self, parent, simulation_energy=None, measurement_energy=None,
         self.spectrum_type = spectrum_type
-        self.show_simulated_sum_spectra = show_simulated_sum_spectra
-        self.show_measured_sum_spectra = show_measured_sum_spectra
-        if self.spectrum_type == SpectrumType.SIMULATION:
-            if self.show_simulated_sum_spectra and len(self.simulation_energy.spectrum_files) > 0:
-                energy_spectra_directory = next(iter(self.simulation_energy.spectrum_files.keys())).parent
-                self.simulated_sum_spectrum = SumEnergySpectrum(self.simulation_energy.spectrum_files,
-                                                                energy_spectra_directory,
-                                                                SumSpectrumType.SIMULATED)
-            if self.show_measured_sum_spectra and len(self.measurement_energy.spectrum_files) > 0:
-                energy_spectra_directory = next(iter(self.measurement_energy.spectrum_files.keys())).parent
-                self.measured_sum_spectrum = SumEnergySpectrum(self.measurement_energy.spectrum_files,
-                                                               energy_spectra_directory,
-                                                               SumSpectrumType.MEASURED)
-            else:
-                self.measured_sum_spectrum = None
-        else:
-            self.simulated_sum_spectrum = None
-        if self.spectrum_type == SpectrumType.MEASUREMENT and \
-                self.show_measured_sum_spectra and len(self.measurement_energy.spectrum_files) > 0:
+        self.simulated_sum_spectrum_is_selected = \
+            simulated_sum_spectrum_is_selected
+        self.measured_sum_spectrum_is_selected = \
+            measured_sum_spectrum_is_selected
+
+        self.measured_sum_spectrum = None
+        self.simulated_sum_spectrum = None
+
+        if self.spectrum_type == SpectrumTab.SIMULATION:
+            """SIMULATION SIMULATION"""
+            if self.simulated_sum_spectrum_is_selected and \
+                    self.simulation_energy:
+                energy_spectra_directory = next(
+                    iter(self.simulation_energy.keys())).parent
+                self.simulated_sum_spectrum = SumEnergySpectrum(
+                    self.simulation_energy,
+                    energy_spectra_directory, SumSpectrumType.SIMULATED)
+            """SIMULATION MEASUREMENT"""
+            if self.measured_sum_spectrum_is_selected and \
+                    self.measurement_energy:
+                # When a sum spectrum is generated for a simulation, both the
+                # simulated and the measured sum spectra should be saved in the
+                # simulation's folder.
+                energy_spectra_directory = next(
+                    iter(self.simulation_energy.keys())).parent
+                self.measured_sum_spectrum = SumEnergySpectrum(
+                    self.measurement_energy,
+                    energy_spectra_directory, SumSpectrumType.MEASURED)
```

Comment:
The idea is correct, but there's an issue: if there's no simulation sum spectrum selected, `self.simulation_energy` is empty. I came up with an alternative solution, I'll push it here.

Also, normal comments work better here than docstrings ("""text""").